### PR TITLE
libjxl: add versoin 0.9.1 and support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **libjxl/***

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libjxl/all/conanfile.py [hunk 1]
```diff
@@ -13,90 +15,112 @@ class LibjxlConan(ConanFile):
...
-        self.requires("lcms/2.11")
+        if Version(self.version) < "0.7.0":
+            self.requires("highway/0.12.2")
+        else:
+            self.requires("highway/1.0.2")
+        self.requires("lcms/2.14")
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
```
Comment (uilianries) [inline recipes/libjxl/all/conanfile.py:61]:
```suggestion
            check_min_cppstd(self, self._minimum_cpp_standard)
        highway = self.dependencies["highway"]
        if not highway.options.shared and self.options.shared:
            raise ConanInvalidConfigution(f"{self.ref}:shared=True requires -o highway:shared=True")
```
To present when forcing that option from other recipes.

File: recipes/libjxl/all/conanfile.py [hunk 2]
```diff
@@ -13,90 +15,115 @@ class LibjxlConan(ConanFile):
...
+        highway = self.dependencies["highway"]
+        if not highway.options.shared and self.options.shared:
```
Comment (uilianries) [inline recipes/libjxl/all/conanfile.py:63]:
```suggestion
        if hasattr(highway, "options.shared") and not highway.options.shared and self.options.shared:
```
Reading highway recipe, older versions has no shared option.

File: recipes/libjxl/all/conanfile.py [hunk 3]
```diff
@@ -13,90 +16,110 @@ class LibjxlConan(ConanFile):
...
+        tc.variables["JPEGXL_ENABLE_MANPAGES"] = False
+        tc.variables["JPEGXL_ENABLE_SJPEG"] = False
+        tc.variables["JPEGXL_ENABLE_OPENEXR"] = False
+        tc.variables["JPEGXL_ENABLE_SKCMS"] = False
+        tc.variables["JPEGXL_ENABLE_TCMALLOC"] = False
+        tc.variables["JPEGXL_FORCE_SYSTEM_BROTLI"] = True
+        tc.variables["JPEGXL_FORCE_SYSTEM_HWY"] = True
+        tc.variables["JPEGXL_FORCE_SYSTEM_LCMS2"] = True
+        if cross_building(self):
+            tc.variables["CMAKE_SYSTEM_PROCESSOR"] = \
```
Comment (SpaceIm) [inline recipes/libjxl/all/conanfile.py:80]: useless with CMakeToolchain

File: recipes/libjxl/all/conanfile.py [hunk 4]
```diff
@@ -13,90 +16,110 @@ class LibjxlConan(ConanFile):
...
+        else:
+            self.requires("highway/1.0.3")
+        self.requires("lcms/2.14")
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        highway = self.dependencies["highway"]
+        if hasattr(highway, "options.shared") and not highway.options.shared and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref}:shared=True requires -o highway:shared=True")
```
Comment (SpaceIm) [inline recipes/libjxl/all/conanfile.py:60]: why is it required?

File: recipes/libjxl/all/conanfile.py [hunk 5]
```diff
@@ -13,90 +16,110 @@ class LibjxlConan(ConanFile):
...
+        libcxx = self.settings.get_safe("compiler.libcxx")
+        if libcxx in ["libstdc++", "libstdc++11"]:
+            return "stdc++"
+        elif libcxx in ["libc++"]:
+            return "c++"
+        elif libcxx in ["c++_shared"]:
+            return "c++_shared"
+        elif libcxx in ["c++_static"]:
+            return "c++_static"
+        return None
```
Comment (SpaceIm) [inline recipes/libjxl/all/conanfile.py:121]: there is stdcpp_library in conan.tools.build

Comment (toge) [reply recipes/libjxl/all/conanfile.py:121]: @SpaceIm 
Sorry for my late response.
I'm going to improve this PR with your comments.

**Comment by RubenRBS** [comment]
Hi @toge thanks a lot for your contribution and sorry that this one feel thru the cracks.

I'm in the process of trying to pick up the older PRs that have been left behind, and this is the first one I got to.

As you have not enabled maintainer permissions to push directly to your fork, could you please apply this patch to the PR? I can build fine on Linux on my end, and the msvc error might have been fixed now (It also bumps some deps and does some minor extra v2 changes):


<details>

```
diff --git a/recipes/libjxl/all/conanfile.py b/recipes/libjxl/all/conanfile.py
index 5a63e3b6e..d16317d54 100644
--- a/recipes/libjxl/all/conanfile.py
+++ b/recipes/libjxl/all/conanfile.py
@@ -1,15 +1,14 @@
 from conan import ConanFile
-from conan.errors import ConanInvalidConfiguration
 from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir
 from conan.tools.build import check_min_cppstd, stdcpp_library
 from conan.tools.scm import Version
 from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
-from conan.tools.microsoft import is_msvc
 
 import os
 
 required_conan_version = ">=1.53.0"
 
+
 class LibjxlConan(ConanFile):
     name = "libjxl"
     description = "JPEG XL image format reference implementation"
@@ -51,18 +50,13 @@ class LibjxlConan(ConanFile):
         if Version(self.version) < "0.7.0":
             self.requires("highway/0.12.2")
         else:
-            self.requires("highway/1.0.6")
+            self.requires("highway/1.0.7")
         self.requires("lcms/2.14")
 
     def validate(self):
         if self.info.settings.compiler.cppstd:
             check_min_cppstd(self, self._minimum_cpp_standard)
 
-        # FIXME: In cci's CI, execution error (error code 3221225781) is raised on msvc in shared build.
-        # This error can't be reproduced outside cci's CI.
-        if is_msvc(self) and self.options.shared:
-            raise ConanInvalidConfiguration(f"{self.name} does not support shared libraries on msvc.")
-
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
@@ -110,7 +104,7 @@ class LibjxlConan(ConanFile):
 
     def package_info(self):
         # jxl
-        self.cpp_info.components["jxl"].names["pkg_config"] = "libjxl"
+        self.cpp_info.components["jxl"].set_property("pkg_config_name", "libjxl")
         self.cpp_info.components["jxl"].libs = [self._lib_name("jxl")]
         self.cpp_info.components["jxl"].requires = ["brotli::brotli",
                                                     "highway::highway",
@@ -119,24 +113,19 @@ class LibjxlConan(ConanFile):
         # in shared build, install jxl only.
         # https://github.com/libjxl/libjxl/blob/v0.5.0/lib/jxl.cmake#L544-L546
         if not self.options.shared:
-            self.cpp_info.components["jxl_dec"].names["pkg_config"] = "libjxl_dec"
+            self.cpp_info.components["jxl_dec"].set_property("pkg_config_name", "libjxl_dec")
             self.cpp_info.components["jxl_dec"].libs = [self._lib_name("jxl_dec")]
             self.cpp_info.components["jxl_dec"].requires = ["brotli::brotli",
                                                             "highway::highway",
                                                             "lcms::lcms"]
 
         # jxl_threads
-        self.cpp_info.components["jxl_threads"].names["pkg_config"] = \
-            "libjxl_threads"
-        self.cpp_info.components["jxl_threads"].libs = \
-            [self._lib_name("jxl_threads")]
-        if self.settings.os == "Linux":
+        self.cpp_info.components["jxl_threads"].set_property("pkg_config_name", "libjxl_threads")
+        self.cpp_info.components["jxl_threads"].libs = [self._lib_name("jxl_threads")]
+        if self.settings.os in ["Linux", "FreeBSD"]:
             self.cpp_info.components["jxl_threads"].system_libs = ["pthread"]
 
         if not self.options.shared and stdcpp_library(self):
-            self.cpp_info.components["jxl"].system_libs.append(
-                stdcpp_library(self))
-            self.cpp_info.components["jxl_dec"].system_libs.append(
-                stdcpp_library(self))
-            self.cpp_info.components["jxl_threads"].system_libs.append(
-                stdcpp_library(self))
+            self.cpp_info.components["jxl"].system_libs.append(stdcpp_library(self))
+            self.cpp_info.components["jxl_dec"].system_libs.append(stdcpp_library(self))
+            self.cpp_info.components["jxl_threads"].system_libs.append(stdcpp_library(self))
```

</details>

**Comment by HappySeaFox** [comment]
Hi, any progress?

**Comment by valgur** [comment]
@toge Feel free to continue working on this PR, but just FYI, the #18812 PR for libjxl has working support for v0.9.1. It's currently only blocked by a required fix (#22380) for the added `gperftools` dependency, which the project tries to use if possible.
I have also added support for all options that the project exposes, plus the patching logic is a bit simpler (imo) and portable across versions.

**Comment by toge** [comment]
@valgur 
Sorry I missed your PR.
My purpose in wrestling with this PR was the fear that no one would maintain libjxl.
Since you will be maintaining it, I can close this PR with peace of mind.
Thanks a lot!

**Pull request closed without merging**